src/geventhttpclient/connectionpool.py

```diff
@@ -204,8 +204,13 @@ def _create_tcp_socket(self, family, socktype, protocol):
                 family, socktype, protocol)
 
             if self.ssl_context_factory is None:
-                ssl_options = self.default_options.copy()
-                ssl_options.update(self.ssl_options)
+                ssl_options = {}
+                if not self.insecure:
+                    ssl_options = self.default_options.copy()
+                    ssl_options.update(self.ssl_options)
```

Comment:
Hm, shouldn't we always respect the self.ssl_options if they were supplied with the insecure flag together? In other words we should probably always copy them over.